lib/govuk_tech_docs/path_helpers.rb

```diff
@@ -0,0 +1,29 @@
+module GovukTechDocs
+  module PathHelpers
+    def get_path_to_resource(config, resource, current_page)
+      if config[:relative_links]
+        resource_path_segments = resource.path.split("/").reject(&:empty?)[0..-2]
+        resource_file_name = resource.path.split("/")[-1]
+
+        path_to_site_root = path_to_site_root config, current_page.path
+        resource_path = path_to_site_root + resource_path_segments
+                                           .push(resource_file_name)
+                                           .join("/")
+      else
+        resource_path = resource.url
+      end
+      resource_path
+    end
+
+    def path_to_site_root(config, page_path)
+      if config[:relative_links]
+        number_of_ascents_to_site_root = page_path.to_s.split("/").reject(&:empty?)[0..-2].length
+        ascents = number_of_ascents_to_site_root.zero? ? ["."] : number_of_ascents_to_site_root.times.collect { ".." }
+        path_to_site_root = ascents.join("/").concat("/")
+      else
+        path_to_site_root = "/"
```

Comment:
Can we include `config[:http_prefix]` here, for cases where the user is using that instead of relative links?

Reply:
Could you give an example? I'm not quite sure what behaviour we'd be after here.

Reply:
I think, [based on what the `middleman-search` library does](https://github.com/manastech/middleman-search/blob/ed9e5254bb554988c755fc40238d9e9f67b20abe/lib/middleman-search/extension.rb#L29), that we just need to have `path_to_site_root` be `config[:http_prefix]`:

```suggestion
        path_to_site_root = config[:http_prefix]
```

I think `:http_prefix` should always be the path to the site root, even if the user hasn't defined it, and it should already default to `/`:

https://github.com/middleman/middleman/blob/97ddbc52eac7a937adcc7d418fa9c2fb4a0a1a1e/middleman-core/lib/middleman-core/application.rb#L154

Reply:
Have just pushed a change to incorporate this.